optimize/backend/src/test/java/io/camunda/optimize/service/util/BpmnModelUtilTest.java

```diff
@@ -0,0 +1,106 @@
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.service.util;
+
+import static io.camunda.optimize.util.ZeebeBpmnModels.END_EVENT;
+import static io.camunda.optimize.util.ZeebeBpmnModels.SERVICE_TASK;
+import static io.camunda.optimize.util.ZeebeBpmnModels.START_EVENT;
+import static io.camunda.optimize.util.ZeebeBpmnModels.USER_TASK;
+import static io.camunda.optimize.util.ZeebeBpmnModels.createSimpleServiceTaskProcess;
+import static io.camunda.optimize.util.ZeebeBpmnModels.createSimpleUserTaskProcess;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.optimize.dto.optimize.FlowNodeDataDto;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.camunda.bpm.model.bpmn.BpmnModelInstance;
+import org.junit.jupiter.api.Test;
+
+public class BpmnModelUtilTest {
+
+  public static final String CUSTOMER_ONBOARDING = "CustomerOnboarding";
+  public static final String END_EVENT_TYPE = "endEvent";
+  public static final String SERVICE_TASK_TYPE = "serviceTask";
+
+  public static final String SIMPLE_SERVICE_TASK_PROCESS =
+      Bpmn.convertToString(createSimpleServiceTaskProcess(CUSTOMER_ONBOARDING));
+
+  @Test
+  void shouldParseBpmnModel() {
+    final BpmnModelInstance modelInstance =
+        BpmnModelUtil.parseBpmnModel(SIMPLE_SERVICE_TASK_PROCESS);
+    assertThat(modelInstance).isNotNull();
+  }
+
+  @Test
+  void shouldExtractFlowNodeData() {
+    final List<FlowNodeDataDto> flowNodeData =
+        BpmnModelUtil.extractFlowNodeData(SIMPLE_SERVICE_TASK_PROCESS);
+    assertThat(flowNodeData).isNotNull();
+    assertThat(flowNodeData.isEmpty()).isFalse();
+    FlowNodeDataDto endEventflowNodeDataDto = flowNodeData.get(0);
+    assertThat(endEventflowNodeDataDto.getId()).isNotNull();
+    assertThat(endEventflowNodeDataDto.getId()).isEqualTo(END_EVENT);
+    assertThat(endEventflowNodeDataDto.getType()).isEqualTo(END_EVENT_TYPE);
+
+    FlowNodeDataDto serviceTaskflowNodeDataDto = flowNodeData.get(1);
+    assertThat(serviceTaskflowNodeDataDto.getId()).isNotNull();
+    assertThat(serviceTaskflowNodeDataDto.getId()).isEqualTo(SERVICE_TASK);
+    assertThat(serviceTaskflowNodeDataDto.getName()).isEqualTo(SERVICE_TASK);
+    assertThat(serviceTaskflowNodeDataDto.getType()).isEqualTo(SERVICE_TASK_TYPE);
+
+    FlowNodeDataDto startEventflowNodeDataDto = flowNodeData.get(2);
+    assertThat(startEventflowNodeDataDto.getId()).isNotNull();
+    assertThat(startEventflowNodeDataDto.getId()).isEqualTo(START_EVENT);
+    assertThat(startEventflowNodeDataDto.getName()).isEqualTo(START_EVENT);
+    assertThat(startEventflowNodeDataDto.getType()).isEqualTo(START_EVENT);
```

Comment:
one of the nice things about assertJ is that you can do fluent assertions. So this whole block could become:
```
assertThat(flowNodeData)
        .isNotNull()
        .isNotEmpty()
        .extracting(FlowNodeDataDto::getId, FlowNodeDataDto::getType, FlowNodeDataDto::getName)
        .containsExactlyInAnyOrder(
            Tuple.tuple(START_EVENT, START_EVENT, START_EVENT),
            Tuple.tuple(SERVICE_TASK, SERVICE_TASK_TYPE, SERVICE_TASK),
            Tuple.tuple(END_EVENT, END_EVENT_TYPE, null));
```

Reply:
fixed